### PR TITLE
railjson: split up TVM into TVM300 and TVM430 signaling systems

### LINKED ISSUE #6282: Separate TVM 300 and TVM 430 signaling systems
### Description and goal

Currently TVM 430 is a signal flag. It should be a separate system altogether.

### Acceptance criteria

 They are separate

### PR DESCRIPTION
**Pull request by Khoyo**

### CODE REVIEW

**Review by multun — APPROVED** [review]
Looks good, couldn't do any testing though. It's probably fine since tvm support is a placeholder anyway

**Comment by Khoyo** [comment]
Indeed, the split is in preparation for implementing it for real

File: core/kt-fast-collections-generator/src/main/kotlin/collections/ArrayList.kt [hunk 1]
```diff
@@ -250,6 +250,16 @@ private fun CollectionItemType.generateArrayList(context: GeneratorContext, curr
                 res.add(c)
                 return res
             }
+
+            /** GENERATED CODE */
+            fun ${paramsDecl} mutable${simpleName}ArrayListOf(a: $type, b: $type, c: $type, d: $type): Mutable${simpleName}ArrayList${paramsUse} {
```
Comment (Erashin) [inline core/kt-fast-collections-generator/src/main/kotlin/collections/ArrayList.kt:255]: Might be possible to use `vararg` here. Would remove the necessity to reimplement this method every time we want to add another argument.

**Review by Erashin — APPROVED** [review]
Core part lgtm

File: core/kt-fast-collections-generator/src/main/kotlin/collections/ArrayList.kt (hunk 1 above)
Comment (Khoyo) [reply core/kt-fast-collections-generator/src/main/kotlin/collections/ArrayList.kt:255]: I opened an issue for that: #6337

**Review by shenriotpro — APPROVED** [review]
OK for tests/ (impl is not yet ready for actual tests)

**Review by flomonster — APPROVED** [review]
LGTM

**Pull request merged**